### PR TITLE
finished front end auth

### PR DESCRIPTION
**Pull request by KChungTM**
## Motivation
Authentication for front end to allow users to navigate through pages. Also fixed some front end aesthetic issues and small bugs. Additionally, cleaned up code!

Issue #29 

## Summary of Changes
- Routing users to login and rerouting if not authenticated.
- Implemented authentication on front end by creating endpoint to check authentication and passing user information.
- Created a getUserProfile endpoint to get emails and privileges.
- Fixed aesthetic issue on proposalDetails because button was smushed.
- Conditionally rendered proposalManagement and Members to not allow non-admin users to view
- Conditionally rendered NavBar
- Fixed proposalCreation money bug where users couldn't put in money value correctly if it contained weird Strings.

## Testing
👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 

## Screenshot/Gif of results
image

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -7,36 +7,67 @@ import ProposalDetails from './pages/proposalDetails/ProposalDetails.js';
...
+
+  const [userID, setUserID] = useState(0);
+  const [privileges, setPrivileges] = useState('Non-Voting Member');
+
+  async function getProfile() {
+    const response = await axios.get('http://localhost:8000/getProfile', {withCredentials: true});
+
+    setUserID(response.data.id);
+    setPrivileges(response.data.privileges);
+  }
```
Comment (ryanhzhao) [inline client/src/App.js]: Consolidate into one function?

File: client/src/App.js [hunk 2]
```diff
@@ -7,36 +7,67 @@ import ProposalDetails from './pages/proposalDetails/ProposalDetails.js';
...
-        </Switch>
+          {doneLoading && <Switch>
+            <Route exact path="/">
+                <Login/>
+            </Route>
+            <Route path="/login">
+              <Login />
+            </Route>
+            <Route path="/dashboard">
+              {isAuth ? <> <NavBar privileges={privileges}/> <Dashboard privileges={privileges}/> </> : <Redirect to="/login"/>}
```
Comment (ryanhzhao) [inline client/src/App.js]: Does the dashboard also need userID to make comments/votes directly from the dashboard?

File: client/src/pages/dashboard/Dashboard.js [hunk 3]
```diff
@@ -52,7 +51,6 @@ function Dashboard() {
     } 
   }
 
-
   const[textboxValue, setTextboxValue] = React.useState('');
 
   const handleSubmit = async () => {
```
Comment (ryanhzhao) [inline client/src/pages/dashboard/Dashboard.js:70]: Update the hardcoded user_id!!

File: client/src/pages/dashboard/Dashboard.js [hunk 4]
```diff
@@ -75,6 +73,7 @@ function Dashboard() {
   };
 
   const submitVote = async (voteDecision) => {
```
Comment (ryanhzhao) [inline client/src/pages/dashboard/Dashboard.js:89]: update the hardcoded user id!! (and remove console.log statements)

File: server/app.js [hunk 5]
```diff
@@ -75,12 +75,30 @@ app.get('/auth/google',
...
+    }
+  } catch (error) {
+    console.log(error.stack);
+  }
+})
+
+app.get('/login',);
+
+
+app.get('')
```
Comment (ryanhzhao) [inline server/app.js]: Remove leftover code

**Review by ryanhzhao — CHANGES_REQUESTED** [review]
Great work on this! I think you guys missed one small section about integrating user_id into votes, but otherwise you're all good!

File: client/src/pages/dashboard/Dashboard.js (hunk 3 above)
Comment (ryanhzhao) [reply client/src/pages/dashboard/Dashboard.js:70]: in handleSubmit()

File: client/src/pages/dashboard/Dashboard.js (hunk 4 above)
Comment (ryanhzhao) [reply client/src/pages/dashboard/Dashboard.js:89]: in submitVote()

**Review by ryanhzhao — APPROVED** [review]

**Review by Warrenwwang1 — APPROVED** [review]

**Pull request merged**